riscos_toolbox/application.py

```diff
@@ -1,15 +1,48 @@
-from .events import EventHandler
+from .events import EventHandler, registered_wimp_events
+from consts import Wimp
 from . import initialise, run
 
 _application = None
 
 
+def _make_poll_flags(events):
+    flags = 0
+
+    if Wimp.Null not in events:
+        flags |= Wimp.Poll.NullMask
+    if Wimp.RedrawWindow not in events:
+        flags |= Wimp.Poll.RedrawWindowRequestMask
+    if Wimp.PointerLeavingWindow not in events:
+        flags |= Wimp.Poll.PointerLeavingWindowMask
+    if Wimp.PointerEnteringWindow not in events:
+        flags |= Wimp.Poll.PointerEnteringWindowMask
+    if Wimp.MouseClick not in events:
+        flags |= Wimp.Poll.MouseClickMask
+    if Wimp.KeyPressed not in events:
+        flags |= Wimp.Poll.KeyPressedMask
+    if Wimp.LoseCaret not in events:
+        flags |= Wimp.Poll.LoseCaretMask
+    if Wimp.GainCartet not in events:
+        flags |= Wimp.Poll.GainCaretMask
+    if Wimp.PollwordNonZero not in events:
+        flags |= Wimp.Poll.PollWordNonZeroMask
+
+    return flags
+
+
 class Application(EventHandler):
     def __init__(self, appdir):
         super().__init__()
         global _application
         _application = self
+        self.poll_flags = _make_poll_flags(registered_wimp_events)
```

Comment:
Does this need updating if you register more events later?

Reply:
It will, but you will have to do that manually. It’s using the decorator to drive the default mask, so I’d you have a handler function setup it will enable it. If you want finer grained control you will have to do it. Perhaps the mask setup should be either an explicit call, or at least you should be able to override it though.

Reply:
That's fair enough.